<?xml version="1.0"?>
<launch>
  <arg name="camera_name" default="front_camera" />
  <arg name="rtsp_url" default="rtsp://admin:#Manabotix27@192.168.131.64:554/Streaming/Channels/101" />
  <arg name="frame_id" default="$(arg camera_name)" />
  <arg name="camera_info_url" default="file:///$(find rtsp_ffmpeg)/config/front_camera_calibration.yaml" />
  <arg name="flip_horizontal" default="false" />
  <arg name="flip_vertical" default="false" />

  <group ns="$(arg camera_name)">
    <node pkg="rtsp_ffmpeg" type="rtsp_ffmpeg" name="$(arg camera_name)_stream" output="screen"> 
      <remap from="camera" to="image_raw" />
      <param name="camera_name" type="string" value="$(arg camera_name)" />
      <param name="rtsp_url" type="string" value="$(arg rtsp_url)" />
      <param name="frame_id" type="string" value="$(arg frame_id)" />
      <param name="camera_info_url" type="string" value="$(arg camera_info_url)" />
      <param name="flip_horizontal" type="bool" value="$(arg flip_horizontal)" />
      <param name="flip_vertical" type="bool" value="$(arg flip_vertical)" />
    </node>
  </group>

  <include file="$(find image_processor)/launch/image_resize.launch" />
</launch>
